src/models/blueprint-collection.js

```diff
@@ -0,0 +1,100 @@
+import path from 'path';
+import walkSync from 'walk-sync';
+import fs from 'fs';
+import _map from 'lodash/map';
+import _filter from 'lodash/filter';
+import _isNil from 'lodash/isNil';
+import _isBool from 'lodash/isBoolean';
+import _isString from 'lodash/isString';
+import _isArray from 'lodash/isArray';
+import _flatten from 'lodash/flatten';
+import _uniq from 'lodash/uniq';
+
+export default class BlueprintCollection {
+  constructor (settings) {
+    this.settings = settings;
+    this.setSearchPaths();
+  }
+
+  allPossiblePaths () {
+    return _flatten(
+      _map(
+        this.settings,
+        (arr, base) => _map(arr, (bp) => expandPath(base, bp) ))
+    );
+  }
+
+  setSearchPaths () {
+    this.searchPaths = _uniq(_filter(this.allPossiblePaths(), validSearchDir));
+  }
+
+  all() {
+    if (this.allBlueprints) {
+      return this.allBlueprints;
+    } else {
+      return this.allBlueprints = this.discoverBlueprints();
+    }
+  }
+
+  addBlueprint(path) {
+    // do we do Blueprint#list and Blueprint#load stuff here
+    // or does the blueprint take it from here?
+    // in any case we need to return an instantiated Blueprint here
+    return path;
+  }
+
```

Comment:
I suggest we keep Blueprint#load the responsibility of Blueprint so it handles the mixin logic. And move any discovery, lookup, list functionality into here.

Reply:
I think you're right.  I'll implement it that way.